vcpkg/users/binarycaching-troubleshooting.md

```diff
@@ -0,0 +1,164 @@
+[DEBUG] </abientries>
+```
+
+You might notice the `triplet_abi` entry has three hashes: the hash of the file content of the `x86-windows` triplet, the `windows.cmake` toolchain, and the compiler hash. Theses hashes would change if you decide to target a different platform.
+
+### The hashes of the files used to build your library (`portfile.cmake`, `*.patch`, or `vcpkg.json`) change between runs
+
+This usually means the baseline of your libraries is not fixed. Frequently, the problem occurs when you clone the latest repository from vcpkg, which will pull in the updated baseline of libraries.
+Possible solutions:
+
+* (recommended) The copy of the vcpkg catalog must be pinned to a specific commit. When you run `git clone vcpkg`, make sure you run `git checkout [commit_id]` or add vcpkg as a submodule in your project. The reason is the ABI of a library is calculated based on the hashes of scripts and ports in the vcpkg repository. Pulling the latest changes will likely cause frequent rebuilds.
+* Set `builtin-baseline` in your [vcpkg manifest](./users/manifests.md).
+
+### The hash of the vcpkg internal CMake functions (vcpkg_from_github, vcpkg_copy_pdbs, vcpkg_fixup_pkgconfig, etc) changes between runs
+
+You are always pulling the latest internal CMake functions, which are used in the ABI calculation. This is probably also caused by cloning the latest vcpkg repo. We expect this issue to go away in the future when we version these functions.
+Possible solutions:
+
+* See the previous section on how to pin the vcpkg catalog to a specific commit.
+
+### The version of CMake, Powershell, compiler version changes between runs
+
+Somehow, your environment changes between runs. To verify that this indeed is the issue, temporarily disable including the compiler hash in your ABI by setting `VCPKG_DISABLE_COMPILER_TRACKING` to `ON` in your custom triplet. If the issue persists, fix tool versions with `--x-abi-tools-use-exact-versions`. Note it is not recommended to use these options in development or a production environment, as they can cause very hard to diagnose errors.
+
+In these cases, vcpkg is behaving correctly, we expect the ABI hashes to change if your toolchain changes. Rebuilds can be expensive, however, so we have recommendations for the following environments:
+
+* CI: We’ve seen this as an issue if you are using CI (such as GitHub Hosted Runners) and using an image that may update your C++/C compiler or CMake version from underneath you. To minimize this, choose a versioned stable image if possible.
+* Visual Studio C++ Compiler Updates: Major, minor and even patch updates to the compiler will cause your library ABI to change.  To minimize this, stop automatic Visual Studio updates.
+
+If the above options do not work, consider the following solutions:
+
+* Use `vcpkg export` to package your libraries instead (currently, only works in classic mode)
```

Comment:
Isn't the export for manifest mode feature available now? Worth double checking.